lib/cocoapods/generator/embed_frameworks_script.rb

```diff
@@ -138,7 +138,7 @@ def script
 
           # Signs a framework with the provided identity
           code_sign_if_enabled() {
-            if [ -n "${EXPANDED_CODE_SIGN_IDENTITY}" -a "${CODE_SIGNING_REQUIRED:-}" != "NO" -a "${CODE_SIGNING_ALLOWED}" != "NO" ]; then
+            if [ -n "${EXPANDED_CODE_SIGN_IDENTITY:-}" -a "${CODE_SIGNING_REQUIRED:-}" != "NO" -a "${CODE_SIGNING_ALLOWED}" != "NO" ]; then
```

Comment:
Have you considered using the one-character-shorter `"${EXPANDED_CODE_SIGN_IDENTITY-}", which only performs the expansion to the empty string if `EXPANDED_CODE_SIGN_IDENTITY` is unset (rather than the empty string or unset, which is what we have here)?